manager/engine.go

```diff
@@ -91,6 +91,14 @@ func (m *VolumeManager) CreateSnapshot(snapshotName string, labels map[string]st
 		return nil, err
 	}
 
+	vol, err := m.ds.GetVolumeRO(volumeName)
+	if err != nil {
+		return nil, err
+	}
+	if vol.Spec.CloneMode == longhorn.CloneModeLinkedClone {
+		return nil, fmt.Errorf("cannot create snapshot for linked-clone volume %v", volumeName)
+	}
```

Comment:
Can we move the check to snapshot/validator.go?

Reply:
I think no because this API talk to engine directly by passing webhook. This is the legacy API